src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionContextExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus.Routing.MessageDrivenSubscriptions
+{
+    using NServiceBus.Routing;
+
+    /// <summary>
+    /// Context extensions for message driven pubsub
+    /// </summary>
+    public static class MessageDrivenSubscriptionContextExtensions
+    {
+        /// <summary>
+        /// Allows the user to specify a explicit address when subscribing
+        /// </summary>
+        public static void UseExplicitPublisherAddress(this SubscribeContext context,string address)
```

Comment:
It's allows the user to send a subscription message to a given address instead of relying on the mappings, sort of similar to `bus.Send(msg, new SendOptions{Address = "myEndpoint"})`. Suggestions for a better name?